python/rtypes/pcc/triggers.py

```diff
@@ -0,0 +1,37 @@
+
+
+class TriggerTime(object):
+    before = 1
+    after = 2
+
+
+class TriggerAction(object):
+    insert = 1
+    read = 2
+    update = 3
+    delete = 4
+
+
+class TriggerProcedure(object):
+    def __init__(self, procedure):
+        self.procedure = procedure
+
+    def __call__(self, *args, **kwargs):
+        return self.procedure(*args, **kwargs)
+
+
+class Trigger(object):
+
+    def __init__(self, pcc_type, time, action, priority=None):
+        self.pcc_type = pcc_type
+        self.time = time
+        self.action = action
+        self.priority = priority
+
+    def __call__(self, procedure):
+        return TriggerProcedure(procedure)
```

Comment:
`return TriggerProcedure(self.pcc_type, self.time, self.action, self.priority, procedure)`
How will dataframe know the type, and other data when it receives the TriggerProcedure object if the information is not present. It does not get the Trigger object. That is just for defining the decorator.